apps/vscode-wing/src/extension.ts

```diff
@@ -15,19 +28,62 @@ export async function activate(context: ExtensionContext) {
     wordPattern: /([a-zA-Z_$][A-Za-z_$0-9]*)/,
   });
 
-  const wingConsoleContext = new WingConsoleManager(context);
-  const languageServerManager = new LanguageServerManager(context);
+  wingConsoleContext = new WingConsoleManager(context);
+  languageServerManager = new LanguageServerManager();
+
+  const wingBinChanged = async () => {
+    Loggers.default.appendLine(`Setting up wing bin...`);
+    const currentWingBin = await getWingBin();
+    if (currentWingBin) {
+      const hasWingBin = await updateStatusBar(currentWingBin);
+
+      wingBinWatcher?.close();
+      wingBinWatcher = watch(
+        currentWingBin,
+        {
+          persistent: false,
+        },
+        async () => {
+          // wait for a second to make sure the file is done moving or being written
+          await new Promise((resolve) => setTimeout(resolve, 1000, undefined));
+
+          await wingBinChanged();
+        }
+      );
+
+      if (hasWingBin) {
+        Loggers.default.appendLine(`Using wing from "${currentWingBin}"`);
+
+        // restart the language server and reset any open console windows
+        languageServerManager?.setWingBin(currentWingBin);
+        wingConsoleContext?.setWingBin(currentWingBin);
+        await languageServerManager?.start();
+        await wingConsoleContext?.stop();
```

Comment:
Would this reset the console's UI state e.g. things the user had typed into the Wing console? If so we could consider prompting the user before reloading everything

Reply:
Good point, will add a prompt